### ISSUE TITLE
Validation time is too long

### ISSUE THREAD

**Issue by yangdaowu** [ws-choi/Conditioned-Source-Separation-LaSAFT#14]
I want to know why the evaluation time is too long. Is this due to GPU issues？ The GPU I am using is a GTX 3090, I checked volatile GPU-Util and found that it is found to be between 80% and 99%
![image](https://user-images.githubusercontent.com/64669901/112075205-038e5b00-8bb3-11eb-9a0e-4d434ddf8d64.png)

**Comment by ws-choi** [comment]
1. The default setting for validation loss is ```raw_l1``` which computes l1 loss between the ground-truth audio signals and the estimated signals. It applies iSTFT to the estimated spectrogram for signal reconstruction, which needs more computational resources.
   - https://github.com/ws-choi/Conditioned-Source-Separation-LaSAFT/blob/88a522604886434ca808ad80d883fd6f00d84572/lasaft/source_separation/conditioned/separation_framework.py#L98
   - If you use ```--val_loss spec_mse``` option, it computes mse loss between spectrograms, which does not require additional iSTFT.

2. For better quality, zero paddings are appended for each instance in eval dataset.

   - https://github.com/ws-choi/Conditioned-Source-Separation-LaSAFT/blob/88a522604886434ca808ad80d883fd6f00d84572/lasaft/data/musdb_wrapper.py#L151

   - Thus, it takes more time.

3. To log separated audio samples, especially for ```wandb``` logging system, it has an additional logic to reconstruct the whole separated sample.

   - https://github.com/ws-choi/Conditioned-Source-Separation-LaSAFT/blob/88a522604886434ca808ad80d883fd6f00d84572/lasaft/source_separation/conditioned/separation_framework.py#L188

   - https://github.com/ws-choi/Conditioned-Source-Separation-LaSAFT/blob/88a522604886434ca808ad80d883fd6f00d84572/lasaft/data/musdb_wrapper.py#L138


4. If you add the ```--check_val_every_n_epoch N``` option, then you can speed up training by skipping some validation phases.

   -  ```--check_val_every_n_epoch 10``` means it computes val_loss for every 10 training epochs.

**Comment by yangdaowu** [comment]
Thanks a lot. I get it @ws-choi
